lib/filewatch/observing_tail.rb

```diff
@@ -65,24 +67,30 @@ def subscribe(observer = NullObserver.new)
     end # def subscribe
 
     private
-    def observe_read_file(path, listener)
-      @buffers[path] ||= FileWatch::BufferedTokenizer.new(@opts[:delimiter])
-      delimiter_byte_size = @opts[:delimiter].bytesize
+    def observe_read_file(watched_file, listener)
       changed = false
       loop do
         begin
-          data = @files[path].sysread(32768)
+          data = watched_file.file_read(32768)
           changed = true
-          @buffers[path].extract(data).each do |line|
+          watched_file.buffer_extract(data).each do |line|
+            @sincedb[watched_file.inode] += (line.bytesize + @delimiter_byte_size)
             listener.accept(line)
-            @sincedb[@statcache[path]] += (line.bytesize + delimiter_byte_size)
```

Comment:
is there a reason to move this? this line is sort of the "ack" after processing. I believe it was purposefully put after the accept to prevent updating the `@sincedb` then suffer some nasty behaviour in accept and essentially skip that line.